integration/base/src/main/java/com/inrupt/client/integration/base/AccessGrantScenarios.java

```diff
@@ -236,252 +235,256 @@ static void teardown() {
+    @DisplayName(":accessGrantQueryByRequestor Lookup Access Grants by requester")
+    void accessGrantQueryByRequesterTest(final Session resourceOwnerSession, final Session requesterSession) {
+        LOGGER.info("Integration Test - Lookup Access Grants by requester");
 
-        final AccessGrantClient accessGrantClient = new AccessGrantClient(
+        final AccessGrantClient requesterAccessGrantClient = new AccessGrantClient(
                 URI.create(ACCESS_GRANT_PROVIDER)
-        ).session(session);
+        ).session(requesterSession);
 
         final Set<String> modes = new HashSet<>(Arrays.asList(GRANT_MODE_READ));
         final Instant expiration = Instant.parse(GRANT_EXPIRATION);
-        final AccessRequest request = accessGrantClient.requestAccess(URI.create(webidUrl),
-            new HashSet<>(Arrays.asList(sharedTextFileURI)), modes, PURPOSES, expiration)
-            .toCompletableFuture().join();
-        final AccessGrant grant = accessGrantClient.grantAccess(request)
-            .toCompletableFuture().join();
-        //Steps
-
-        //1. call verify endpoint to verify grant
-        final var grantVerification = accessGrantClient.verify(grant).toCompletableFuture().join();
-        assertTrue(grantVerification.getChecks().size() > 0);
-        assertEquals(grantVerification.getErrors().size(), 0);
-        assertEquals(grantVerification.getWarnings().size(), 0);
+        final AccessRequest request = requesterAccessGrantClient.requestAccess(URI.create(requesterWebidUrl),
+                        new HashSet<>(Arrays.asList(sharedTextFileURI)), modes, PURPOSES, expiration)
+                .toCompletableFuture().join();
 
-        //2. use the grant to access the target container.
-        //3. try using the grant to access a child resource.
-    }
+        final AccessGrantClient resourceOwnerAccessGrantClient = new AccessGrantClient(
+                URI.create(ACCESS_GRANT_PROVIDER)
+        ).session(resourceOwnerSession);
+        final AccessGrant grant = resourceOwnerAccessGrantClient.grantAccess(request)
+                .toCompletableFuture().join();
 
-    // Query access grant related tests
-    @ParameterizedTest
-    @MethodSource("provideSessions")
-    @DisplayName(":accessGrantQueryByRequestor Lookup Access Grants by requestor")
-    void accessGrantQueryByRequestorTest(final Session session) {
-        LOGGER.info("Integration Test - Lookup Access Grants by requestor");
+        //query for all grants with a dedicated purpose
+        final AccessCredentialQuery<AccessGrant> query = AccessCredentialQuery.newBuilder()
+                .creator(URI.create(requesterWebidUrl)).build(AccessRequest.class);
+        final List<AccessGrant> grants = resourceOwnerAccessGrantClient.query(query).toCompletableFuture().join();
+        assertTrue(grants.size() >= 1); //on a second run there would be 2 grants because grants are never deleted
```

Comment:
This seems like a test that could easily hide an underlying problem. It might be better to scope the query to a resource url so that you will never end up with more than one result.

Reply:
This is solved now.